streams/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -297,6 +311,80 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         .map(_ must beRight(()))
     }
 
+  private def effectAsync =
+    prop { list: List[Int] =>
+      val s = ZStream.effectAsync[Any, Throwable, Int] { k =>
+        list.foreach(a => k(Task.succeed(a)))
+      }
+
+      slurp(s.take(list.size)) must_=== Success(list)
+    }
+
+  private def effectAsyncM = {
+    val list = List(1, 2, 3)
+    unsafeRun {
+      for {
+        latch <- Promise.make[Nothing, Unit]
+        fiber <- ZStream
+                  .effectAsyncM[Any, Throwable, Int] { k =>
+                    latch.succeed(()) *>
+                      Task.succeedLazy {
+                        list.foreach(a => k(Task.succeed(a)))
+                      }
+                  }
+                  .take(list.size)
+                  .run(Sink.collectAll[Int])
+                  .fork
+        _ <- latch.await
+        s <- fiber.join
+      } yield s must_=== list
+    }
+  }
+
+  private def effectAsyncMaybeSome =
+    prop { list: List[Int] =>
+      val s = ZStream.effectAsyncMaybe[Any, Throwable, Int] { _ =>
+        Some(Stream.fromIterable(list))
+      }
+
+      slurp(s.take(list.size)) must_=== Success(list)
+    }
+
+  private def effectAsyncMaybeNone =
+    prop { list: List[Int] =>
+      val s = ZStream.effectAsyncMaybe[Any, Throwable, Int] { k =>
+        list.foreach(a => k(Task.succeed(a)))
+        None
+      }
+
+      slurp(s.take(list.size)) must_=== Success(list)
+    }
+
+  private def effectAsyncInterruptLeft = unsafeRun {
+    for {
+      streamCancelled <- Ref.make[Boolean](false)
+      latch           = internal.OneShot.make[Unit]
+      fiber <- ZStream
+                .effectAsyncInterrupt[Any, Nothing, Unit] { _ =>
+                  latch.set(()); Left(streamCancelled.set(true))
```

Comment:
Can you use a promise instead of `OneShot` here and run it with `nonFlaky` from `RTSSpec` (copy `nonFlaky` into this file and run it locally) just to check if it is indeed flaky?

Reply:
Thanks @iravid I have put in the equivalent test from the RTSSpec which uses a promise, and it seems to be more reliable :-)